emcc.py

```diff
@@ -1591,14 +1591,23 @@ def phase_setup(options, state, newargs):
     settings.DISABLE_EXCEPTION_CATCHING = 0
 
   if settings.WASM_EXCEPTIONS:
-    if 'DISABLE_EXCEPTION_CATCHING' in user_settings:
-      exit_with_error('DISABLE_EXCEPTION_CATCHING is not compatible with -fwasm-exceptions')
-    if 'DISABLE_EXCEPTION_THROWING' in user_settings:
-      exit_with_error('DISABLE_EXCEPTION_THROWING is not compatible with -fwasm-exceptions')
+    # DISABLE_EXCEPTION_CATCHING is 1 by default. So if it is 0, it means the
+    # user specifies it explicitly
+    if not settings.DISABLE_EXCEPTION_CATCHING:
+      exit_with_error('DISABLE_EXCEPTION_CATCHING=0 is not compatible with -fwasm-exceptions')
+    # DISABLE_EXCEPTION_THROWING is 0 by default, but in Wasm EH, it defaults to
+    # 1. If the user explicitly gives DISABLE_EXCEPTION_THROWING=0, error out.
+    if 'DISABLE_EXCEPTION_THROWING' in user_settings and user_settings['DISABLE_EXCEPTION_THROWING'] == '0':
```

Comment:
I think you can just do `user_settings.get('DISABLE_EXCEPTION_THROWING') == '0'` here?

Reply:
Should these two `if` conditions be checking similar things?  Why is one checking `user_settings` but one not?

Reply:
Hmm, didn't know `user_settings.get` doesn't error out but rather just returns nothing when the key does not exist. Switched to `user_settings.get` here and other similar places and removed some unnecessary comments.